src/transformers/generation/utils.py

```diff
@@ -1087,6 +1087,25 @@ def _validate_model_class(self):
                 exception_message += f" Please use one of the following classes instead: {generate_compatible_classes}"
             raise TypeError(exception_message)
 
+    def _validate_assistant(self, assistant_model):
+        if assistant_model is None or not self.config.is_encoder_decoder:
+            return
+
+        if not assistant_model.config.is_encoder_decoder:
+            attributes_to_check = ["encoder_attention_heads", "encoder_ffn_dim", "encoder_layers"]
+            attributes_to_check = [attr for attr in dir(self.config) if attr in attributes_to_check]
+            are_equal = all(
+                getattr(self.config, attr) == getattr(assistant_model.config, attr) for attr in attributes_to_check
+            )
+            if not are_equal:
+                raise ValueError(
+                    "The main model and the assistant don't have encoders of the same size. "
```

Comment:
```suggestion
                    "The main model and the assistant don't have compatible encoder-dependent input shapes. "
```
Suggestion: the assistant is decoder-only, so saying "...assistant don't have encoders of the same size" doesn't make much sense :D I'd reword it into something that points towards the encoder-dependent inputs.